qiskit/providers/aer/pulse_new/de/DE_Problems.py

```diff
@@ -127,6 +129,190 @@ def __init__(self,
+            interval: Time interval.
+            frame: Frame to solve in.
+            cutoff_freq: Cutoff frequency to use when solving.
+        """
+
+        generator = OperatorModel(operators=[-1j * Operator(op) for
+                                             op in hamiltonian._operators],
+                                  signals=hamiltonian.signals,
+                                  signal_mapping=hamiltonian.signal_mapping,
+                                  frame=hamiltonian.frame,
+                                  cutoff_freq=hamiltonian.cutoff_freq)
+
+        super().__init__(generator=generator,
+                         y0=y0,
+                         t0=t0,
+                         interval=interval,
+                         frame=frame,
+                         cutoff_freq=cutoff_freq)
+
+class LindbladProblem(BMDE_Problem):
+    """Simulate a density matrix according to the Lindblad equation:
+
+    .. math::
+        \dot{\rho}(t) = -i[H(t), \rho(t)] + \sum_j \gamma_j(t) L_j\rho(t)L_j^\dagger - \frac{1}{2}\{L_j^\daggerL_j, \rho(t)\}
+
+    where:
+        - :math:`H(t)` is the Hamiltonian,
+        - :math:`L_j` are the noise operators, or dissipators, and
+        - :math:`\gamma_j(t)` are the time-dependent dissipator coefficients,
+    specified in terms of a :class:`QuantumSystemModel` object.
+    """
+
+    def __init__(self,
+                 q_model: QuantumSystemModel,
+                 y0: Optional[np.ndarray] = None,
+                 t0: Optional[float] = None,
+                 interval: Optional[List[float]] = None,
+                 frame: Optional[Union[str, Operator, np.ndarray, BaseFrame]] = 'auto',
+                 cutoff_freq: Optional[float] = None):
+        """Constructs a BMDE_Problem representing the Lindblad equation,
+        to act on a density matrix.
+
+        Additionally, 'frame' may be specified in this class either as a
+        standard anti-Hermitian operator :math:`F`, or as a Hermitian
+        operator :math:`H`, in which case it enters the frame :math:`F=-iH`.
+
+        Args:
+            hamiltonian: The Hamiltonian to simulate.
+            y0: Initial state, intended to be a density matrix.
+            t0: Initial time.
+            interval: Time interval.
+            frame: Frame to solve in, specified in terms of the original
+                   equation.
+            cutoff_freq: Cutoff frequency to use when solving.
+        """
+
+        generator = q_model.vectorized_lindblad_generator
+
+        # handle frame
+        # if frame is in hamiltonian, set the generator frame to the vectorized
+        # version of it
+        if q_model.hamiltonian.frame.frame_operator is not None:
+            generator.frame = vec_commutator(q_model.hamiltonian.frame.frame_operator)
+
+            # if a frame was additionally specified as an argument, warn
+            if frame != 'auto':
+                warn("""A frame was specified in both the QuantumSystemModel
```

Comment:
Yeah you basically summarized it: the point is to provide a mechanism for the following use cases:
- The user wants to work in the lab frame, but simulate behind the scenes, for numerical purposes, in a different frame, potentially with cutoff frequencies. In this case, the original model the user holds shouldn't be modified by these transformations, and results should be returned in the lab frame.
- The user is fully working in a frame (i.e. the model has a frame), and they want to solve and get results in that frame. In this case they can define the problem using a model that is already in the frame of interest. The problem detects this, and then sets an internal flag to simulate in that frame and also return results in that frame. 

This functionality is important but I do feel like this implementation is slightly awkward. The problems and this logic were written before the `Frame` objects, so maybe using those this could be simplified (maybe in a subsequent PR). Perhaps we can change the problem arguments to be something like: 
- change the name of the problem `frame` argument to `solve_in_frame` (making explicit that this is a frame of the solver), and 
- add a `results_in_frame` kwarg, specifying which frame to return results in (which would default to the frame the model is in).
- similarly, we could change `cutoff_freq` to `solve_with_cutoff_freq` or something like this, to again make explicit that this is purely a property of the simulation, not of the model object the user is working with